File: scratch/gopurs-adt-reuse-validation-20260916/after-pure/src/Test/RecordsFFICheatcode.cc
#include "purescript.h"
namespace {
struct Inner { int e, f; };
struct Middle { int c; Inner d; };
struct Record { int a; Middle b; };
}
FOREIGN_BEGIN( Test_RecordsFFICheatcode )
exports["runRecordsFFICheatcode"] = [](const boxed& in) -> boxed {
    Record record{0, {0, {0, 0}}};
    for (int n = unbox<int>(in); n > 0; --n) {
        record.a += 1;
        record.b.c += 2;
        record.b.d.e += 3;
        record.b.d.f += n % 5;
    }
    return record.b.d.f;
};
FOREIGN_END

File: scratch/gopurs-adt-reuse-validation-20260916/before-pure/src/Test/RecordsFFICheatcode.php
<?php
$exports['runRecordsFFICheatcode'] = function($limit) {
    $n = (int)$limit;
    $r = (object)["a" => 0, "b" => (object)["c" => 0, "d" => (object)["e" => 0, "f" => 0]]];
    while ($n > 0) {
        $r->a += 1;
        $r->b->c += 2;
        $r->b->d->e += 3;
        $r->b->d->f += ($n % 5);
        $n--;
    }
    return $r->b->d->f;
};
return $exports;
